modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/SingleMenu.ts

```diff
@@ -104,25 +105,46 @@ export const createAutocompleteItems = (
   const renderIcons = !renderText || MenuUtils.menuHasIcons(items);
   return Optionals.cat(
     Arr.map(items, (item) => {
-      if (item.type === 'separator') {
-        return InlineContent.createSeparatorItem(item).fold(
-          MenuUtils.handleError,
-          (d) => Optional.some(MenuItems.separator(d))
-        );
-      } else {
-        return InlineContent.createAutocompleterItem(item).fold(
-          MenuUtils.handleError,
-          (d: InlineContent.AutocompleterItem) => Optional.some(MenuItems.autocomplete(
-            d,
-            matchText,
-            renderText,
-            'normal',
-            onItemValueHandler,
-            itemResponse,
-            sharedBackstage,
-            renderIcons
-          ))
-        );
+      switch (item.type) {
+        case 'separator':
+          return InlineContent.createSeparatorItem(item).fold(
+            MenuUtils.handleError,
+            (d) => Optional.some(MenuItems.separator(d))
+          );
+
+        case 'cardmenuitem':
+          return BridgeMenu.createCardMenuItem(item).fold(
+            MenuUtils.handleError,
+            (d) => Optional.some(MenuItems.card(
+              {
+                ...d,
+                // Intercept action
+                onAction: (_api) => onItemValueHandler(d.value, d.meta)
```

Comment:
As discussed, we should probably still call the original `onAction` still if it was defined and then fire the autocomplete on action.